cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -1252,6 +1252,143 @@ TriangleMesh TriangleMesh::SelectByIndex(const core::Tensor &indices) const {
     return result;
 }
 
+namespace {
+
+/// This is a templatized local version of RemoveDuplicates.
+/// Use of templates allows us to use common code for different
+/// datatype possibilities for vertex.positions, namely, float32, float64,
+/// and triangle.indices, namely, int32, int64.
+/// This function first computes two maps
+/// (MapA.) From vertex coordinates to their indices.
+//          This is implemented using unordered_map.
+/// (MapB.) From the vertex indices before duplicate removal to indices after
+/// removal.
+///          This is implemented using std::vector.
+///
+/// MapA allows the function to detect duplicates and MapB allows it update
+/// triangle indices. During the computation of the above maps unique vertex
+/// attributes including positions are moved to beginning of the corresponding
+/// attribute tensor.
+///
+/// Caveats:
+///   The unordered_map approach is fast in general but if the inputs are close,
+///   the hashing could cause problems and it might be good to consider a
+///   sorting based algorithms, which can do a better job of comparing the
+///   double values.
+/// \param mesh The mesh from which duplicate vertices are to be removed.
+template <typename Coord_t, typename Tindex_t>
+TriangleMesh &RemoveDuplicateVerticesWorker(TriangleMesh &mesh) {
+    // Algorithm based on Eigen implementation
+    if (!mesh.HasVertexPositions()) {
+        utility::LogWarning(
+                "TriangeMesh::RemoveDuplicateVertices: No vertices present, "
+                "ignoring.");
+        return mesh;
+    }
+
+    typedef std::tuple<Coord_t, Coord_t, Coord_t> Point3d;
+    std::unordered_map<Point3d, size_t, utility::hash_tuple<Point3d>>
+            point_to_old_index;
+
+    auto vertices = mesh.GetVertexPositions().To(mesh.GetDevice()).Contiguous();
+    auto orig_num_vertices = vertices.GetLength();
+    const auto vmask_type = core::Bool;
+    using vmask_itype = bool;
```

Comment:
Maybe a personal taste, but I find the original types, bool and core::Bool, more descriptive in this case.